pkg/costmodel/allocation.go

```diff
@@ -375,6 +375,68 @@ func (cm *CostModel) ComputeAllocation(start, end time.Time, resolution time.Dur
 	podPVCMap := map[podKey][]*PVC{}
 	buildPodPVCMap(podPVCMap, pvMap, pvcMap, podMap, resPodPVCAllocation)
 
+	// Because PVCs can be shared among pods, the respective PV cost
+	// needs to be evenly distributed to those pods based on time
+	// running, as well as the amount of time the PVC was shared.
+
+	// Build a relation between every PVC to the pods that mount it
+	// and a window representing the interval during which they
+	// were associated.
+	pvcPodIntervalMap := make(map[pvcKey]map[podKey]kubecost.Window)
+
+	for _, pod := range podMap {
+
+		for _, alloc := range pod.Allocations {
```

Comment:
That's correct. This is essentially just the code which we already were using to get PV hours, just moved into its own loop because I needed that info before setting the cost.

Looking back on it, I see that it's definitely unnecessary to go through every allocation, though to get to `alloc.Properties` I would still need to at least look at one.

Reply:
This should now only look at a single allocation within `pod.Allocations`. Not an enormous change, but accessing the keys of `pod.Allocation` would take an additional loop over containers to begin with, so this just cuts down the number of iterations of this loop to one.